kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,463 @@
+            }
+        }
+    }
+
+    /**
+     * Perform application of ServiceAccount, Roles and CRDs needed for proper cluster operator deployment.
+     * Configuration files are loaded from kroxylicious-operator directory.
+     */
+    public void applyClusterOperatorInstallFiles(String namespaceName) {
+        for (File operatorFile : getFilteredOperatorFiles(installFiles)) {
+            final String resourceType = operatorFile.getName().split("\\.")[1];
+
+            if (resourceType.equals(Constants.NAMESPACE)) {
+                Namespace namespace = ReadWriteUtils.readObjectFromYamlFilepath(operatorFile, Namespace.class);
+                if (!NamespaceUtils.isNamespaceCreated(namespace.getMetadata().getName())) {
+                    try {
+                        kubeClient().getClient().load(new FileInputStream(operatorFile.getAbsolutePath())).create();
+                    }
+                    catch (FileNotFoundException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                }
+            }
+            else {
+                try {
+                    kubeClient().getClient().load(new FileInputStream(operatorFile.getAbsolutePath()))
+                            .inNamespace(namespaceName)
+                            .create();
+                }
+                catch (FileNotFoundException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Prepare environment for cluster operator which includes creation of namespaces, custom resources and operator
+     * specific config files such as ServiceAccount, Roles and CRDs.
+     * @param clientNamespace namespace which will be created and used as default by kube client
+     * @param namespaces list of namespaces which will be created
+     */
+    public void prepareEnvForOperator(String clientNamespace, List<String> namespaces) {
+        applyCrds();
+        applyClusterOperatorInstallFiles(clientNamespace);
+
+        if (cluster.cluster().isOpenshift() && kubeClient().getNamespace(Environment.KROXY_ORG) != null) {
+            for (String namespace : namespaces) {
+                LOGGER.debug("Setting group policy for Openshift registry in Namespace: {}", namespace);
+                Exec.exec(Arrays.asList("oc", "policy", "add-role-to-group", "system:image-puller", "system:serviceaccounts:" + namespace, "-n", Environment.KROXY_ORG));
+            }
+        }
+    }
+
+    /**
+     * Temporary method to fulfill the Crds installation until new JOSDK 5.0.0 release landed https://github.com/operator-framework/java-operator-sdk/releases
+     */
+    private void applyCrds() {
+        String path = Constants.PATH_TO_OPERATOR + "/src/main/resources/META-INF/fabric8/";
+        List<File> crdFiles = Arrays.stream(Objects.requireNonNull(new File(path).listFiles())).sorted()
```

Comment:
whats the point of the `Objects.requireNonNull`?

Reply:
Assure that the path is correct. If something change in the location of the CRDs, this check would notify us.

`ListFiles()` retuns `null` if the abstract pathname does not denote a directory

Reply:
I think that's rather obscure. An explicit java.io.File#isDirectory with a clear error message would be better.